lib/puppet/pops/types/p_runtime_type.rb

```diff
@@ -93,19 +93,19 @@ def from_puppet_name(puppet_name)
   # a class that is the same or subclass of t1's resolved runtime type name
   # @api private
   def _assignable?(o, guard)
-    return false unless o.is_a?(PRuntimeType)
+    return false unless o.instance_of?(PRuntimeType)
```

Comment:
I think we want to keep `is_a?(PRuntimeType)` so we return `false` if `o` is a subclass